src/common/ceph_json.cc

```diff
@@ -675,9 +277,9 @@ long long JSONFormattable::val_long_long() const {
 
 bool JSONFormattable::val_bool() const {
   return (boost::iequals(value.str, "true") ||
+	  boost::iequals(value.str, "1") ||
           boost::iequals(value.str, "on") ||
-          boost::iequals(value.str, "yes") ||
-          boost::iequals(value.str, "1"));
+          boost::iequals(value.str, "yes"));
 }
 
 string JSONFormattable::def(const string& def_val) const {
```

Comment:
I'll make another pass to get these inlined and ditto for a number of the other functions (find* especially) in here; additional plans to get string_view to work efficiently with the containers in the implementation via transparent comparitor.